docs/spec/WIP_provisioning/overview.md

```diff
@@ -0,0 +1,229 @@
+   change), 
+ - when a validator chooses to change the commission on fees, all accumulated 
+   commission fees must be simultaneously withdrawn.
+
+When the validator is the proposer of the round, that validator (and their
+delegators) receives between 1% and 5% of fee rewards, the reserve tax is then
+charged, then the remainder is distributed socially by voting power to all
+validators including the proposer validator.  The amount of proposer reward is
+calculated from pre-commits Tendermint messages. All provision rewards are
+added to a provision reward pool which validator holds individually. Here note
+that `BondedShares` represents the sum of all voting power saved in the
+`GlobalState` (denoted `gs`).
+
+```
+proposerReward = feesCollected * (0.01 + 0.04 
+                  * sumOfVotingPowerOfPrecommitValidators / gs.BondedShares)
+validator.ProposerRewardPool += proposerReward
+
+reserveTaxed = feesCollected * params.ReserveTax
+gs.ReservePool += reserveTaxed
+
+distributedReward = feesCollected - proposerReward - reserveTaxed
+gs.FeePool += distributedReward
+gs.SumFeesReceived += distributedReward
+gs.RecentFee = distributedReward
+```
+
+The entitlement to the fee pool held by the each validator can be accounted for
+lazily.  First we must account for a validator's `count` and `adjustment`. The
+`count` represents a lazy accounting of what that validators entitlement to the
+fee pool would be if there `VotingPower` was to never change and they were to
+never withdraw fees. 
+
+``` 
+validator.count = validator.VotingPower * BlockHeight
+``` 
+
+Similarly the GlobalState count can be passively calculated whenever needed,
+where `BondedShares` is the updated sum of voting powers from all validators.
+
+``` 
+gs.count = gs.BondedShares * BlockHeight
+``` 
+
+The `adjustment` term accounts for changes in voting power and withdrawals of
+fees. The adjustment factor must be persisted with the validator and modified
+whenever fees are withdrawn from the validator or the voting power of the
+validator changes. When the voting power of the validator changes the
+`Adjustment` factor is increased/decreased by the cumulative difference in the
+voting power if the voting power has been the new voting power as opposed to
+the old voting power for the entire duration of the blockchain up the previous
+block. Each time there is an adjustment change the GlobalState (denoted `gs`)
+`Adjustment` must also be updated.
+
+```
+simplePool = validator.count / gs.count * gs.SumFeesReceived
+projectedPool = validator.PrevPower * (height-1) 
+                / (gs.PrevPower * (height-1)) * gs.PrevFeesReceived
+                + validator.Power / gs.Power * gs.RecentFee
+
+AdjustmentChange = simplePool - projectedPool
+validator.AdjustmentRewardPool += AdjustmentChange
+gs.Adjustment += AdjustmentChange
+```
+
+Every instance that the voting power changes, information about the state of
```

Comment:
"*Whenever* the voting power changes, ...."

Reply:
WIP, another PR